### PR TITLE
Update the go version to 1.22

### PR DESCRIPTION
**Pull request by surola**
Updated the go version to 1.22 (latest) to fix the critical vulnerabilities

### CODE REVIEW

**Comment by yermulnik** [comment]
Fixes: https://github.com/warrensbox/terraform-switcher/issues/325
Is blocked by: https://github.com/warrensbox/terraform-switcher/issues/324

**Review by MatrixCrawler — APPROVED** [review]
Change seems simple enough. Approved.

but we could consider switching from circleci to github workflows.

**Review by warrensbox — APPROVED** [review]

**Comment by warrensbox** [comment]
I merged to master. I will be testing this on the master branch before merging it to the release branch (which will create an official release)

**Pull request merged**
